app/services/service_offering/add_to_portfolio_item.rb

```diff
@@ -0,0 +1,38 @@
+module ServiceOffering
+  class AddToPortfolioItem
+    def initialize(params)
+      @req_params = params
+    end
+
+    def process
+      # Get the api response object from Topology
+      TopologicalInventory.call do |api_instance|
+        @service_offering = api_instance.show_service_offering(@req_params[:service_offering_ref])
+      end
+
+      params = HashWithIndifferentAccess.new
```

Comment:
@lindgrenj6 - Testing a method this large will be difficult.  I think it would be beneficial to break this method up into the specific parts.  `def process` can be the caller of those methods, but breaking out behavior would help testing and troubleshooting.

Reply:
There are a number of services that require a call to the TopologicalInventory, there is precedent in those services to duplicate the procedures used in those services here as well.

Reply:
Done @syncrou , see my latest commit with tests included.